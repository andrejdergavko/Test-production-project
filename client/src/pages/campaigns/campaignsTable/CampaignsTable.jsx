// @flow
import * as React from "react";
import { Table } from "baseui/table-semantic";
import { useStyletron } from "styletron-react";

import Spinner from "./spinner";

const COLUMNS = [
  "Campaign name",
  "Channels",
  "Created At",
  "Clicks",
  "Views",
  "Created By",
  "Status",
  "",
];

type CampaignsTableRowT = (string | React.Node)[];

type CampaignsTableT = {
  data: CampaignsTableRowT[],
  loading?: boolean,
};

function CampaignsTable({ data, loading }: CampaignsTableT) {
  const [css] = useStyletron();
  const tableBox = css({
    position: "relative",
    minHeight: "200px",
  });

  return (
    <div className={tableBox}>
      {loading && <Spinner />}
      <Table
        columns={COLUMNS}
        data={data}
        overrides={{
          Root: {
            style: () => {
              return { border: "#f6f6f6 solid 2px" };
            },
          },
          TableHeadCell: {
            style: () => {
              return {
                backgroundColor: "#f6f6f6",
                "::before": { borderLeftStyle: "none" },
                "::after": { backgroundImage: "none" },
              };
            },
          },
          TableBodyRow: {
            style: () => {
              return {
                outline: "#f6f6f6 solid 1px",
                backgroundColor: "#fff",
              };
            },
          },
          TableBodyCell: {
            style: () => {
              return {
                verticalAlign: "middle",
              };
            },
          },
        }}
      />
    </div>
  );
}

export default CampaignsTable;
