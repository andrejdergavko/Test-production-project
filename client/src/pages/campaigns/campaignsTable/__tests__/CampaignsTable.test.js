import React from "react";
import renderer from "react-test-renderer";

import CampaignsTable from "../CampaignsTable";

describe("CampaignsTable component:", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<CampaignsTable />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
