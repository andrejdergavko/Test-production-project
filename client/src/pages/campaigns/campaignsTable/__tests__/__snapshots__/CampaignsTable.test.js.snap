// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampaignsTable component: renders correctly 1`] = `
<div
  className=""
  data-baseweb="table-semantic"
>
  <table
    className=""
  >
    <thead
      className=""
    >
      <tr
        className=""
      >
        <th
          className=""
        >
          Campaign name
        </th>
        <th
          className=""
        >
          Channels
        </th>
        <th
          className=""
        >
          Created At
        </th>
        <th
          className=""
        >
          Clicks
        </th>
        <th
          className=""
        >
          Views
        </th>
        <th
          className=""
        >
          Created By
        </th>
        <th
          className=""
        >
          Status
        </th>
        <th
          className=""
        >
          
        </th>
      </tr>
    </thead>
    <tbody
      className=""
    >
      <tr
        className=""
      >
        <td
          className=""
        />
        <td
          className=""
        />
        <td
          className=""
        />
        <td
          className=""
        />
        <td
          className=""
        />
        <td
          className=""
        />
        <td
          className=""
        />
        <td
          className=""
        />
      </tr>
    </tbody>
  </table>
</div>
`;
